### PR TITLE
51: Add control over exchanges and queue bindings to the broker, remo…

### PR DESCRIPTION
**Pull request by lfse-slafleur**
…ve the 'ReceiveNextMessage' consumer in favor of adding a single message_threshold number in QueueSubscription and add support for multiple SDKs in the AvailableWorkflow feature.

After merging, do not forget to:
- [ ] Update optimizer worker
- [ ] Update simulator worker
- [ ] Update orchestrator
  - [ ] https://github.com/Project-OMOTES/orchestrator/pull/56
  - [ ] https://github.com/Project-OMOTES/orchestrator/issues/57
- [ ] Update example SDKs in omotes-system

### CODE REVIEW

File: src/omotes_sdk/internal/common/broker_interface.py [hunk 1]
```diff
@@ -348,46 +393,20 @@ def remove_queue_subscription(self, queue_name: str) -> None:
             self._remove_queue_subscription(queue_name), self._loop
         ).result()
 
-    def receive_next_message(
```
Comment (lfse-slafleur) [inline src/omotes_sdk/internal/common/broker_interface.py:351]: Removed as we merged `QueueSubscriptionConsumer` and `QueueSingleMessageConsumer`. The single messages functionality is now maintained by setting `delete_after_messages=1`

File: src/omotes_sdk/internal/worker/worker.py [hunk 2]
```diff
@@ -57,6 +57,7 @@ def update_progress(self, fraction: float, message: str) -> None:
             message,
         )
         self.broker_if.send_message_to(
+            None,
```
Comment (lfse-slafleur) [inline src/omotes_sdk/internal/worker/worker.py:60]: Use default exchange

File: src/omotes_sdk/omotes_interface.py [hunk 3]
```diff
@@ -94,30 +94,36 @@ class OmotesInterface:
...
         self.broker_if.start()
+        self.broker_if.declare_exchange(OmotesQueueNames.omotes_exchange_name())
         self.connect_to_available_workflows_updates()
         self.request_available_workflows()
 
-        while not self._workflow_config_received.is_set():
-            logger.info("Waiting for workflow definitions to be received from the orchestrator...")
-            self._workflow_config_received.wait(timeout=5)
+        logger.info("Waiting for workflow definitions to be received from the orchestrator...")
+        if not self._workflow_config_received.wait(timeout=60):
```
Comment (cwang39403) [inline src/omotes_sdk/omotes_interface.py:121]: Do we want to set this as a configurable variable?

**Review by cwang39403 — APPROVED** [review]

**Review by MarkTNO — APPROVED** [review]
Looks good!

File: src/omotes_sdk/omotes_interface.py (hunk 3 above)
Comment (lfse-slafleur) [reply src/omotes_sdk/omotes_interface.py]: Good one! Fixed: https://github.com/Project-OMOTES/omotes-sdk-python/pull/53/commits/bbea78f85d9095e092abe087f041f827831637fe

**Pull request merged**